pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -353,3 +467,59 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 	log.Debugf("cluster flavor %q not recognised. Defaulting to false", c.clusterFlavor)
 	return false
 }
+
+// GetVolumeAnnotations fetches annotations from PVC bound to passed volumeID and returns
+// annotation key-value pairs as a map
+func (c *K8sOrchestrator) GetVolumeAnnotations(ctx context.Context, volumeID string) (map[string]string, error) {
+	log := logger.GetLogger(ctx)
+	log.Debugf("Getting volume annotations for volume: %s", volumeID)
+	if pvc, ok := c.volumeIDToPvcMap[volumeID]; ok {
+		parts := strings.Split(pvc, "/")
+		pvcNamespace := parts[0]
+		pvcName := parts[1]
+
+		pvcObj, err := c.k8sClient.CoreV1().PersistentVolumeClaims(pvcNamespace).Get(ctx, pvcName, metav1.GetOptions{})
```

Comment:
Why are we not using PVCListers() in c.informerManager ? This will be a faster lookup instead of querying the api server

Reply:
Done.